### ISSUE TITLE
Player can levitate with Walk Over/Under Monsters

### ISSUE THREAD

**Issue by retro65** [fabiangreffrath/crispy-doom#703]
### Background

Version of Crispy Doom: 5.10.0 (6d81ad8567d)

Operating System and version: Ubuntu 18.04.5 LTS

Game: Doom

Any loaded WADs and mods (please include full command line):

[floatbug.zip](https://github.com/fabiangreffrath/crispy-doom/files/5720813/floatbug.zip)
Contains the test map, which includes a `DEHACKED` lump to make the Arch-Vile slower and unable to damage.
And a ~700 KB MP4 video of me performing the bug, since it cannot be done when recording demos (physical crispness disabled).

### Bug description

Steps:
0. Have Walk Over/Under Monsters enabled.
1. Enable the full automap with things visible (`IDDT` twice), it will make positioning easier.
2. Turn on run.
3. Alert the Arch-Vile
4. Wait until the Arch-Vile is attacking and close enough to jump onto, get on him.
5. Turn off run and open automap.
6. Wait until you hear the attack sound (to ensure he's still), then sneak so the tail of Doomguy's arrow is aligned with the tip of the Arch-Vile's triangle, then stop.
7. Done. Now Archie is moving again, and as long as you don't move, you stay mid-air. If Recoil Thrust is disabled, you can even fire weapons without resetting the bug.

Observed behavior: If player is not moving horizontally, gravity isn't applied when the monster the player was standing on has moved. In the recording, as soon as I tap a movement key Doomguy begins to fall, not sooner.

Expected behavior: Player should fall..? Hard to say what exactly is the "correct" behavior since vanilla doesn't have walking over-under things.

**Comment by fabiangreffrath** [comment]
Yes, I know, this is also the reason why https://github.com/fabiangreffrath/crispy-doom/issues/381 isn't marked as fixed yet. :frowning: Vertical movement is still a bit borked with this feature. So, thanks for submitting your use case and the test level, so maybe I'll get this fixed some day.
